### PR TITLE
Feature/22 change resource reference data type to list

### PR DESCRIPTION
**Pull request by lovoll**
Make place for storing many ResourceReferences

## Description
As we need to store metadata for a Resource the datatype for ResourceReference is changed from ReourceReference? to List<ReourceRefernce> this makes it possible to store MAskinportenIDentifier and all claims defined fro the Resource.

Also added rsourcetypes for DelegationSchemeIdentifier and DelegationSchemeScope

## Related Issue(s)
- #22

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs [hunk 1]
```diff
@@ -17,6 +17,18 @@ public enum ReferenceType : int
         OrgApp = 2,
 
         [EnumMember(Value = "Uri")]
-        Uri = 3
+        Uri = 3,
+        
+        [EnumMember(Value = "DelegationSchemeId")]
+        DelegationSchemeId = 4,
+
+        [EnumMember(Value = "DelegationScope")]
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs:25]: ServiceCodeVersion and OrgApp are temp values created during POC of the components and can be removed.
DelegationScope should perhaps be more specifically named for MaskinportenScope.

The ResourceType for these resources has also been decided to rename from api-schema to maskinporten-schema to be more specific. Which will need update in Enum both for the resource registry and in your migration job.

Comment (lovoll) [reply src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs]: Pushed changes to reflect comment and updated the Export job to include ResourceType as this was added after the code was copied and sets the correct value on export.

File: src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs [hunk 2]
```diff
@@ -10,13 +10,19 @@ public enum ReferenceType : int
         [EnumMember(Value = "Default")]
         Default = 0,
 
-        [EnumMember(Value = "ServiceCodeVersion")]
-        ServiceCodeVersion = 1,
+        [EnumMember(Value = "Uri")]
+        Uri = 3,
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs:14]: ResourceType is also defined as a Type in the database and will need updated deploy script

File: src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs [hunk 3]
```diff
@@ -10,13 +10,19 @@ public enum ReferenceType : int
...
+        
+        [EnumMember(Value = "DelegationSchemeId")]
+        DelegationSchemeId = 4,
 
-        [EnumMember(Value = "OrgApp")]
-        OrgApp = 2,
+        [EnumMember(Value = "MaskinportenScope")]
+        DelegationScope = 5,
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs:20]: Mismatch between Attribute and Property naming.

File: src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs (hunk 2 above)
Comment (jonkjetiloye) [reply src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs:14]: Should numbering be reordered also? Might break peoples local test data. Nothing created in AT environments yet I think

Comment (lovoll) [reply src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs]: Done

File: src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs (hunk 3 above)
Comment (lovoll) [reply src/Altinn.ResourceRegistry.Core/Enums/ReferenceType.cs]: Fixed

**Review by jonkjetiloye — APPROVED** [review]

**Pull request merged**